frontend/packages/erd-core/src/features/erd/components/ERDContent/ERDContent.tsx

```diff
@@ -70,11 +77,38 @@ export const ERDContentInner: FC<Props> = ({
   const {
     state: { loading },
   } = useERDContentContext()
-  const { isTableGroupEditMode } = useUserEditingStore()
+  const { isTableGroupEditMode, hoverInfo } = useUserEditingStore()
   const { tableName: activeTableName } = useUserEditingActiveStore()
 
   const { selectTable, deselectTable } = useTableSelection()
 
+  useEffect(() => {
+    updateNodesAndEdges(hoverInfo)
+  }, [hoverInfo])
+
+  const updateNodesAndEdges = (hoverInfo: HoverInfo) => {
+    const filteredEdges =
+      hoverInfo.columnName !== undefined && hoverInfo.columnType
+        ? _edges.filter((edge) => {
+            if (hoverInfo.columnName === 'id') {
+              const edgSourceId = `${hoverInfo.tableName}_id`
+              return edge.id.includes(edgSourceId)
+            }
+            return edge.targetHandle?.split('-')[1] === hoverInfo.columnName
+          })
+        : _edges
+    const { edges: updatedEdges } = highlightNodesAndEdges(
+      nodes,
+      filteredEdges,
+      {
+        activeTableName,
+        hoverTableName: hoverInfo.tableName,
+      },
+    )
+
+    setEdges(updatedEdges)
+  }
```

Comment:
Perhaps it can be expressed with only the information that `TableColumn` has now.



https://github.com/liam-hq/liam/blob/56572e24025c947fe31a1c2ef3f2f43f1015d4b7/frontend/packages/erd-core/src/features/erd/components/ERDContent/components/TableNode/TableColumnList/TableColumn/TableColumn.tsx#L13-L14

When `isSource: true`, it is a primary key, and when `targetCardinality` has a value, it is a foreign key.
Therefore, if `isSelectedTable && (isSource || !!targetCardinality)`, it would be better to change the background color.